### ISSUE TITLE
misc/cgo/testplugin: TestIssue25756pie broken on darwin/arm64 on release-branch.go1.17

### ISSUE THREAD

**Issue by bcmills** [golang/go#52995]
```
--- FAIL: TestIssue25756pie (1.25s)
    plugin_test.go:274: ./issue25756pie.exe: exit status 2
        panic: plugin.Open("life"): plugin was built with a different version of package sync/atomic
```

`greplogs -l -e 'plugin was built with a different version of package sync/atomic' --since=2022-01-01`
[2022-05-18T16:46:28-0e7138a/darwin-arm64-11](https://build.golang.org/log/2fcfe8c7dc7719728200b58cd5723c112eee3c3f)
[2022-05-18T16:46:28-0e7138a/darwin-arm64-12](https://build.golang.org/log/94c8aea69a1552510ddfb2c0279792449e00767e)

―

Curiously, this failed on the `darwin-arm64-11` and `darwin-arm64-12` builders but not `darwin-arm64-11_0-toothrot`. 🤔
I wonder if the new `arm64` builders are misconfigured in some way?

(This is a release-blocker because `darwin/arm64` is a first class port; attn @golang/release.)

**Comment by bcmills** [comment]
It looks like the test is using the `go` command from `$PATH` rather than the one from `GOROOT/bin`:
https://cs.opensource.google/go/go/+/master:misc/cgo/testplugin/plugin_test.go;l=116;drc=afd181cf0b69c3591d7e47ceca4fabf14434d77e

That could have something to do with it.

(That will be fixed automatically by [CL 404134](https://golang.org/cl/404134), but that isn't even merged at head, let alone at 1.17.)

**Comment by bcmills** [comment]
I can't reproduce this on a `gomote` instance. (I tried both `go test ./cgo/testplugin` and `go tool dist test testplugin` using a `gomote push` of the current `release-branch.go1.17`.)

I suspect that this may be a bad interaction with some bug in the `buildlet` binary on this builder, and possibly also a remaining bug in the test (https://golang.org/cl/407315 didn't fix it).

From the output of `go version -m` I determined that it is from some time at or after CL [356589](https://golang.org/cl/356589) but before CL [381735](https://golang.org/cl/381735) (Jan. 28). Notably, it does not include CL [354754](https://golang.org/cl/354754).

As a next step I suggest that we bump the [`buildletVersion`](https://cs.opensource.google/go/x/build/+/master:cmd/buildlet/buildlet.go;l=85;drc=b92283589b8badaff34afceab830890c94a98acc) constant in `cmd/buildlet` and deploy a fresh build of it to these builders.

**Comment by cherrymui** [comment]
See also #46239. It is not reproducible locally or on gomote. The skip is specific to the toothrot builders https://cs.opensource.google/go/go/+/release-branch.go1.17:misc/cgo/testplugin/plugin_test.go;l=268

**Comment by bcmills** [comment]
Just a hunch: https://go.dev/cl/353449 was included in Go 1.18, but not backported to 1.17. That might interact with `dist test` and the buildlet in some non-trivial way to produce what appears to be version skew due to a mismatched `GOROOT`.

I'll send a backport of that through the SlowBot to see if it fixes things. (https://go.dev/cl/407881)

**Comment by bcmills** [comment]
Looks like that's the jackpot!
https://storage.googleapis.com/go-build-log/cefcdb71/darwin-arm64-12_6a7eca57.log
